user-interface/src/components/Chart/index.js

```diff
@@ -55,49 +43,25 @@ export default class Chart extends Component {
       return;
     }
 
-    // Format data, then see how much to scroll by
-    const tempData = this._formatData(newProps.data);
-    this.dataPoints += tempData[0].length - 1;
-    const scrollPos = this.dataPoints - this.props.maxPoints;
-
-    // Add the data and scroll
-    this.chart.flow({ columns: tempData, duration: 500, to: scrollPos });
+    this.scrollPos += 1;
+    this.chart.flow({ json: [newProps.data], keys: { value: ['humidity', 'temperature', 'ec'] }, duration: 500, to: this.scrollPos });
   }
 
-  // Takes the current props and transforms them to the C3 data format
-  _formatData(data) {
-    const sensorData = data[this.props.sensorName];
-    const tempData = {};
-
-    // Iterate through all the data points
-    for (let i = 0; i < sensorData.length; i++) {
-      for (const key in sensorData[i]) {
-        if (!(key in tempData)) {
-          tempData[key] = [];
-        }
-
-        tempData[key].push(sensorData[i][key]);
-      }
-    }
-
-    // Create new chart data
-    const newChartData = [];
-    for (const key in tempData) {
-      newChartData.push([key].concat(tempData[key]));
-    }
-
-    return newChartData;
-  }
+  scrollPos = -this.props.maxPoints;
+  chart = null
+  dataPoints = 0
 
   // Enable/disable real-time data
-  handleClick() {
+  handleClick = () => {
     this.setState({
```

Comment:
Consider providing this.setState() a function instead of an object. Since react will batch multiple setState() calls, there's an edge case that the previous state may not be what you expect.

```
this.setState(prevState => ({active: !prevState.active}));
```